homeassistant/components/klyqa/config_flow.py

```diff
@@ -0,0 +1,307 @@
+
+    async def init(self) -> None:
+        """Initialize."""
+        if self.inited:
+            return
+        self.inited = True
+        integration_data, cached = await api.async_json_cache(
+            None, "last.klyqa_integration_data.cache.json"
+        )
+        if cached:
+
+            self.username = str(integration_data[CONF_USERNAME])
+            self.password = str(integration_data[CONF_PASSWORD])
+            self.scan_interval = int(integration_data[CONF_SCAN_INTERVAL])
+            self.host = str(integration_data[CONF_HOST])
+            self.sync_rooms = bool(integration_data[CONF_SYNC_ROOMS])
+            self.polling = bool(integration_data[CONF_POLLING])
+
+    def get_klyqa(self) -> HAKlyqaAccount | NoneType:
+        """Get Klyqa account."""
+        if self.klyqa:
+            return self.klyqa
+        if (
+            not self.hass
+            or DOMAIN not in self.hass.data
+            or not self.hass.data[DOMAIN].klyqa_accounts
+            or not self.hass.data[DOMAIN].klyqa_accounts
+        ):
+            return None
+        self.klyqa = self.hass.data[DOMAIN].klyqa_accounts[0]
+        return self.klyqa
+
+    async def _show_setup_form(
+        self, errors: dict[Any, Any] | None = None
+    ) -> FlowResult:
+        """Show the setup form to the user."""
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME, default=self.username): cv.string,
+                    vol.Required(CONF_PASSWORD, default=self.password): cv.string,
+                    vol.Required(CONF_POLLING, default=self.polling): bool,
+                    vol.Required(CONF_SCAN_INTERVAL, default=self.scan_interval): int,
+                    vol.Required(CONF_SYNC_ROOMS, default=self.sync_rooms): bool,
+                    vol.Required(CONF_HOST, default=self.host): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if self.get_klyqa():
+            # already logged in from platform or other way
+            return self.async_abort(reason="single_instance_allowed")
+
+        await self.init()
+
+        login_failed = False
+
+        if user_input is None or login_failed:
+            return await self._show_setup_form()
+
+        self.username = str(user_input[CONF_USERNAME])
+        self.password = str(user_input[CONF_PASSWORD])
+        self.scan_interval = int(user_input[CONF_SCAN_INTERVAL])
+        self.sync_rooms = bool(user_input[CONF_SYNC_ROOMS])
+        self.polling = bool(user_input[CONF_POLLING])
+        self.host = str(user_input[CONF_HOST])
```

Comment:
Instead of storing everything in class variables, you could also just use this data directly here (or pass it along).